src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -76,14 +104,14 @@ class P5VisualizerTemplate extends P5Visualizer(paramDesc) {
         super(seq)
     }
 
-    checkParameters(params: ParamValues<typeof paramDesc>) {
-        const status = super.checkParameters(params)
-
-        // make sure the step size is positive
-        if (params.stepSize <= 0)
-            status.addError('Step size must be positive')
-
-        return status
+    async presketch() {
+        // === Asynchronous setup ===
+        // If any pre-computations must be run before the sketch is created,
+        // placing them in the `preSketch()` function will allow them
+        // to run asynchronously, i.e. without blocking the browser.
+        // The sketch will not be created until this function completes.
+        // By default, intializing the sequence occurs here.
+        super.presketch()
```

Comment:
You should add the `await` before super.presketch(), since that is usually what one needs to do. And then I suggest putting a comment after that to the effect that a time-consuming initialization could go there, but if you don't have anything to do, there's no need to have a presketch() function at all -- no need to define one just to await super.presketch(). This "empty" one is just here as a placeholder, to give an anchor for the explanation.

Reply:
done

Reply:
I took the liberty of amplifying the remarks somewhat, hope that's OK. Resolving.